MODLOGIN-61 Initial module setup

dmytrotur opened the pull request:
MODLOGIN-61

- Initial module setup.
- Added stubs for API calls.

Commit:
Initial module setup

Commit:
Readme fix

id-jenkins commented on src/main/java/org/folio/rest/impl/MessageDeliveryImpl.java:
```diff
@@ -0,0 +1,23 @@
+                                  Map<String, String> okapiHeaders,
+                                  Handler<AsyncResult<Response>> asyncResultHandler,
+                                  Context vertxContext) {
+    //TODO replace stub response
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

id-jenkins commented:
SonarQube analysis reported 1 issue
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

id-jenkins commented:
<html><body><pre>
WARNING: lint-raml-cop: No configuration found for repository 'mod-sender'
WARNING: lint-raml-cop: See FOLIO-903. Add an entry to api.yml
WARNING: lint-raml-cop: Attempting default configuration.
INFO: lint-raml-cop: Investigating mod-sender/ramls
INFO: lint-raml-cop: Using submodule ramls/raml-util at 1d92b1079e865cd5830f49d73c21a9a27f1800e0
INFO: lint-raml-cop: Looking for JSON schema files: ./ramls
WARNING: lint-raml-cop: Configured file not found: dummy.raml
WARNING: lint-raml-cop: Missing from configuration: mod-sender.raml
INFO: lint-raml-cop: Assessing schema files:
INFO: lint-raml-cop: deliveryChannel.json: Each property "description" is present.
ERROR: lint-raml-cop: message.json: Missing "description" for: attachments, channels
ERROR: lint-raml-cop: attachment.json: Missing "description" for: description, disposition, data, contentType, name, contentId
INFO: lint-raml-cop: Assessing RAML files:
INFO: lint-raml-cop: Processing RAML v1.0 file: mod-sender.raml
INFO: lint-raml-cop:   raml-cop did not detect any errors with mod-sender.raml
ERROR: lint-raml-cop: There were processing errors. See list above.
</pre><body></html>

dcrossleyau reviewed (approved):
Nice work.

Use vertx 3.5.4 ... see todays change to mod-rmb-template

Add concise Description at top of front page at GitHub. I added a basic one at dev.folio.org/source-code/ and will modify that when yours is ready.

Link to issue tracker from README. See mod-notes example. If not yet ready then just use general "FOLIO".

descriptors/ModuleDescriptor-template.json
the "id" in "provides" section should not use "artifactId" (multiple interfaces are possible)

Naming of raml files: mod-sender.raml should be just sender.raml

Commit:
Moving to vertx v3.5.4, changing README, raml file renaming

id-jenkins commented:
<html><body><pre>
INFO: lint-raml-cop: Investigating mod-sender/ramls
INFO: lint-raml-cop: Using submodule ramls/raml-util at 1d92b1079e865cd5830f49d73c21a9a27f1800e0
INFO: lint-raml-cop: Looking for JSON schema files: ./ramls
INFO: lint-raml-cop: Assessing schema files:
INFO: lint-raml-cop: deliveryChannel.json: Each property "description" is present.
ERROR: lint-raml-cop: message.json: Missing "description" for: attachments, channels
ERROR: lint-raml-cop: attachment.json: Missing "description" for: contentType, contentId, description, name, data, disposition
INFO: lint-raml-cop: Assessing RAML files:
INFO: lint-raml-cop: Processing RAML v1.0 file: sender.raml
INFO: lint-raml-cop:   raml-cop did not detect any errors with sender.raml
ERROR: lint-raml-cop: There were processing errors. See list above.
</pre><body></html>

dmytrotur commented:
@dcrossleyau , thanks a lot for review.
Fixes:
- moved to vertx v3.5.4
- added concise description. Now it is "Sending messages through delivery channels"
- added link to FOLIO issue tracker
- changed interface id to 'message-delivery' in ModuleDescriptor-template.json
- renamed raml file to 'sender.raml'
